src/aliceVision/depthMap/cuda/commonStructures.hpp

```diff
@@ -145,7 +145,7 @@ template <class Type, unsigned Dim> class CudaHostMemoryHeap
     sz = 1;
     if (Dim >= 1) sx = _size[0];
     if (Dim >= 2) sy = _size[1];
-    if (Dim >= 3) sx = _size[2];
```

Comment:
@griwodz size of the z axis for 3D arrays was always 1 and size of the x axis was wrong.
Sounds surprising to not have more problems than that...

Reply:
The only reason why this didn't lead to horrible crashes is that the Dim>=3 variation is used once in the code, and never these functions.